libraries/Microsoft.Bot.StreamingExtensions/ReceiveRequest.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    /// <summary>
+    /// An incoming request from a remote client.
+    /// </summary>
+    public class ReceiveRequest
+    {
```

Comment:
This entire "Rebuild HTTP on top of WebSockets" is... really werid.

Reply:
This was the whole point of streaming extensions and why this is significant and non trivial.